sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -714,6 +720,132 @@ private String toPrettyJson(Object object) {
+        }
+
+        if (Short.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).shortValue());
+        }
+
+        if (c == Integer.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).intValue();
+        }
+
+        if (c == Integer.TYPE || Integer.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).intValue());
+        }
+
+        if (c == Long.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).longValue();
+        }
+
+        if (Long.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).longValue());
+        }
+
+        if (c == Float.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).floatValue();
+        }
+
+        if (Float.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).floatValue());
+        }
+
+        if (c == Double.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).doubleValue();
+        }
+
+        if (Double.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).doubleValue());
+        }
+
+        if (BigDecimal.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).decimalValue());
+        }
+
+        if (BigInteger.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).bigIntegerValue());
+        }
+
+        if (String.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).asText());
+        }
+
+        if (c == Boolean.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).booleanValue();
+        }
+
+        if (Boolean.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).booleanValue());
+        }
+
+        if (List.class.isAssignableFrom(c)) {
+            try {
+                return OBJECT_MAPPER.readValue(node.toString(), c);
+            } catch (IOException e) {
+                throw new IllegalStateException("Failed to convert to collection.", e);
+            }
+        }
+        if (JsonNode.class.isAssignableFrom(c) || ObjectNode.class.isAssignableFrom(c)) {
+            // JsonNode
+            if (JsonNode.class != c) {
+                if (ObjectNode.class != c) {
+                    throw new IllegalArgumentException(
```

Comment:
a little confused by the exception message, do not supports "subclasses except object node"?

Reply:
Exception existed before (in other helper methods) - basically it means you can't use TextNode or IntegerNode as the generic Type T in container.queryItems - just JsonNode or ObjectNode (you can then of course in you app convert JsonNode into whatever subnode you want to use)

Reply:
make sense, thanks